dspace-api/src/main/java/org/dspace/workflowbasic/dao/impl/BasicWorkflowItemDAOImpl.java

```diff
@@ -37,27 +40,36 @@ protected BasicWorkflowItemDAOImpl() {
 
     @Override
     public BasicWorkflowItem findByItem(Context context, Item i) throws SQLException {
-        Criteria criteria = createCriteria(context, BasicWorkflowItem.class);
-        criteria.add(Restrictions.eq("item", i));
-        // Look for the unique WorkflowItem entry where 'item_id' references this item
-        return uniqueResult(criteria);
+        CriteriaBuilder criteriaBuilder = getCriteriaBuilder(context);
+        CriteriaQuery criteriaQuery = getCriteriaQuery(criteriaBuilder, BasicWorkflowItem.class);
+        Root<BasicWorkflowItem> basicWorkflowItemRoot = criteriaQuery.from(BasicWorkflowItem.class);
+        criteriaQuery.select(basicWorkflowItemRoot);
+        criteriaQuery.where(criteriaBuilder.equal(basicWorkflowItemRoot.get(BasicWorkflowItem_.item), i));
+        return uniqueResult(context, criteriaQuery, false, BasicWorkflowItem.class, -1, -1);
     }
 
     @Override
     public List<BasicWorkflowItem> findBySubmitter(Context context, EPerson ep) throws SQLException {
-        Criteria criteria = createCriteria(context, BasicWorkflowItem.class);
-        criteria.createAlias("item", "i");
-        criteria.add(Restrictions.eq("i.submitter", ep));
-        criteria.addOrder(Order.asc("workflowitemId"));
-        return list(criteria);
+
+
+        CriteriaBuilder criteriaBuilder = getCriteriaBuilder(context);
+        CriteriaQuery criteriaQuery = getCriteriaQuery(criteriaBuilder, BasicWorkflowItem.class);
+        Root<BasicWorkflowItem> basicWorkflowItemRoot = criteriaQuery.from(BasicWorkflowItem.class);
+        Join<BasicWorkflowItem, Item> join = basicWorkflowItemRoot.join("item");
+        criteriaQuery.select(basicWorkflowItemRoot);
+        criteriaQuery.where(criteriaBuilder.equal(join.get(Item_.submitter), ep));
+        return list(context, criteriaQuery, false, BasicWorkflowItem.class, -1, -1);
```

Comment:
This query looks to have lost the `ORDER BY workflowitemID ASC` which exists in current code.